### ISSUE TITLE
[cpprestsdk][OpenSSL] OpenSSL::SSL-NOTFOUND when compiling for RelWithDebInfo

### ISSUE THREAD

**Issue by marcbertola** [microsoft/vcpkg#8822]
**Describe the bug**
I'm compiling a program that depends on cpprestsdk, which has OpenSSL as a dependency.
When I run CMake for Release or Debug (including vcpkg.cmake as a toolchain file), everything works.
When I run CMake for the RelWithDebInfo configuration, it complains about being unable to find OpenSSL::SSL (a target pulled in by cpprestsdk)

Am I doing something that is unsupported? I don't see this problem with any other dependencies compiled under vcpkg.

**Environment**
- OS: Linux x64, Ubuntu 18.04 with CMake 3.15.x
- Compiler: GCC 7.4.0

**To Reproduce**
Triplet file: `x64-linux-dynamic.cmake`

```cmake
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)

set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```

1. ./vcpkg install cpprestsdk::x64-linux-dynamic
2. Run CMake on my project:
`cmake -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo -DVCPKG_TARGET_TRIPLET=x64-linux-dynamic -DCMAKE_TOOLCHAIN_FILE=path_to_scripts/buildsystems/vcpkg.cmake project_dir`
3. `ninja`
4. Error message: `ninja: error: 'OpenSSL::SSL-NOTFOUND', needed by '(my program)', missing and no known rule to make it`

**Expected behavior**
I was hoping that a Release target could be swapped in instead of complaining about the missing RelWithDebInfo configuration. I believe this is what happens with the other dependencies.

I noticed that OpenSSL seems to get special treatment under Linux -- the share/OpenSSL directories don't contain actual target/config files so maybe that has something to do with it?

**Failure logs** 
See error message above, which is the only error output I can see.

**Comment by Neumann-A** [comment]
`set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO "RelWithDebInfo;Release;")` in toolchain or via command line
see PR #6393 but it is totally a cmake setup issue

**Comment by marcbertola** [comment]
Ah, I see. I didn't know you could provide more than one option to that variable.
I originally tried -DCMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO=Release but that didn't work.

Using both, like in your example, did work. Thanks. Hopefully, this will help others who are still getting their head around this stuff.

**Comment by Neumann-A** [comment]
>Using both, like in your example, did work. Thanks. Hopefully, this will help others who are still getting their head around this stuff.

Its actually three RelWithDebInfo, Release and "". Having the last one is important for configuration independent imported targets (as discussed in the mentioned PR).

**Comment by Neumann-A** [comment]
I think your quotes are wrongly placed

```
cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=/home/mbertola/code/vcpkg/scripts/buildsystems/vcpkg.cmake "-DCMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO=RelWithDebInfo;Release;" "-DCMAKE_MAP_IMPORTED_CONFIG_MINSIZEREL=MinSizeRel;Release;" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DVCPKG_TARGET_TRIPLET=x64-linux-dynamic ..
n
```
try
```
cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=/home/mbertola/code/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO="RelWithDebInfo;Release;" -DCMAKE_MAP_IMPORTED_CONFIG_MINSIZEREL="MinSizeRel;Release;" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DVCPKG_TARGET_TRIPLET=x64-linux-dynamic ..
n
```

**Comment by marcbertola** [comment]
Thanks for the suggestion. I tried that too, and the result is the same.

I continued my experiments, removing the portions that depend on cpprestsdk and openssl, but now the problem has shifted to another dependency. So I guess this isn't specific to cpprestsdk or openssl. 

The packages it does seem to stumble on are the ones that don't have any cmake files under vcpkg/installed/x64-linux-dynamic/shared ... in both cases the directory only contains a `vcpkg_abi_info.txt` file, and no cmake config files. Am I correct in understanding that this means that these packages are not built directly by vcpkg, but rather rely on the package being present via   the system package manager (apt install)?

**Comment by Neumann-A** [comment]
if the share folder does not contain anything just means that find_package call will use the CMake integrated `Find<Module>.cmake` files (or user provided ones)

try setting 
`set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO "RelWithDebInfo;Release;")`
at the beginning of your main cmakelists.txt.

I have the feeling that either the empty "" is not passed correctly to cmake or there is some case sensitivity going on between `RelWithDebInfo` and `RELWITHDEBINFO`. 

Also make sure that the build directory is clean of any old cmakecache information

**Comment by marcbertola** [comment]
Good call.
In my top-level CMakeLists.txt, there was some residual code that I had tried before I contacted this list.  It forced those variables back to "Release" only.

- I removed those lines, and used the cmake command line only (now that it doesn't get overridden by my CMakeLists.txt file anymore), but this fails on yet another dependency (cjson).
- However, if I remove the command line options, and put the correct "RelWithDebInfo;Release;" lines in my top-level CMakeLists.txt file, it works!

(I deleted the build directory before each experiment).

So for now, I will keep the settings in my CMakeLists.txt and omit them from the command line until I have time to dig a bit further. 

Thank you very much for your persistent assistance, Alexander. I really appreciate it.
